### PR TITLE
chore: schematics - improve dependency management

### LINKED ISSUE #11592: Schematics - update dependency handling
Currently we have hardcoded list and versions of dependencies in schematics files. From 4.0 we should pull them directly from our root package.json file, so everything is in sync and easier to maintain. (applies to feature libs if those install special external deps not required by core/storefrontlib)

### PR DESCRIPTION
**Pull request by znikola**
closes #11592

### CODE REVIEW

File: projects/schematics/src/shared/utils/package-utils.ts [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
+): NodeDependency[] {
+  const dependencies: NodeDependency[] = [];
+  for (const dependencyName in dependencyObject) {
+    if (!dependencyObject.hasOwnProperty(dependencyName)) {
+      continue;
+    }
+
+    for (const scope of skipScopes) {
+      if (dependencyName.startsWith(scope)) {
+        continue;
```
Comment (marlass) [inline projects/schematics/src/shared/utils/package-utils.ts]: I think this continue won't skip the parent loop (only this scopes loop). You would need to label top level loop: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/label

File: projects/schematics/src/shared/utils/package-utils.ts [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
 import { version } from '../../../package.json';
-import { ANGULAR_CORE, DEFAULT_ANGULAR_VERSION, UTF_8 } from '../constants';
+import { ANGULAR_SCHEMATICS, SPARTACUS_SCOPE, UTF_8 } from '../constants';
 import { getServerTsPath } from './file-utils';
 import { getDefaultProjectNameFromWorkspace } from './workspace-utils';
 
+export const SKIP_SCOPES_FEATURES_LIBS = [SPARTACUS_SCOPE, ANGULAR_SCHEMATICS];
+export function createDependencies(
+  dependencyObject: any,
+  skipScopes: string[] = [SPARTACUS_SCOPE]
```
Comment (marlass) [inline projects/schematics/src/shared/utils/package-utils.ts]: Can we change the default according to usage? So default would be the `[SPARTACUS_SCOPE, ANGULAR_SCHEMATICS]` and we would override it only for main schematics?

File: tools/schematics/dependency-collector.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import { execSync } from 'child_process';
+import fs from 'fs';
+
+interface Dependency {
+  [packageName: string]: string;
+}
+const fileName = 'projects/schematics/src/dependencies.json';
+const packageJsonDirectories: string[] = [
+  'feature-libs',
+  'integration-libs',
```
Comment (marlass) [inline tools/schematics/dependency-collector.ts]: I think we can skip those, just for optimization ;)

File: projects/schematics/src/add-ssr/index.ts [hunk 4]
```diff
@@ -42,11 +42,16 @@ import {
...
 import {
-  getAngularVersion,
   getSpartacusSchematicsVersion,
   readPackageJson,
 } from '../shared/utils/package-utils';
 
+const DEPENDENCY_NAMES: string[] = [
+  '@angular/platform-server',
+  NGUNIVERSAL_EXPRESS_ENGINE,
+  'ts-loader',
```
Comment (marlass) [inline projects/schematics/src/add-ssr/index.ts:52]: So it's really needed?

**Review by marlass — CHANGES_REQUESTED** [review]

File: projects/schematics/src/add-ssr/index.ts (hunk 4 above)
Comment (znikola) [reply projects/schematics/src/add-ssr/index.ts:52]: Not sure. It does compile without it, but I'd rather have @dunqan double check it.

Comment (znikola) [reply projects/schematics/src/add-ssr/index.ts:52]: And it's not really in the scope of this ticket, I'd say :)

File: projects/schematics/src/add-ssr/index.ts [hunk 5]
```diff
@@ -147,39 +152,35 @@ function modifyAppModuleFile(): Rule {
...
+    type: NodeDependencyType.Default,
+    version: spartacusVersion,
+    name: SPARTACUS_SETUP,
+  });
+
+  const thirdPartyDependencies: NodeDependency[] = [];
+  for (const dependencyName of DEPENDENCY_NAMES) {
+    thirdPartyDependencies.push({
+      type: NodeDependencyType.Default,
+      version: (collectedDependencies.storefrontapp as any)[dependencyName],
```
Comment (marlass) [inline projects/schematics/src/add-ssr/index.ts]: `any`? Can't we manage something better?

**Review by marlass — APPROVED** [review]

**Pull request merged**